hivemind/p2p/p2p_daemon_bindings/control.py

```diff
@@ -93,6 +103,99 @@ async def listen(self) -> AsyncIterator["ControlClient"]:
         async with server:
             yield self
 
+    async def _read_from_persistent_conn(self, reader: asyncio.StreamReader):
+        while True:
+            resp: p2pd_pb.Response = p2pd_pb.Response()  # type: ignore
+            await read_pbmsg_safe(reader, resp)
+
+            if resp.HasField("callUnaryResponse"):
+                call_id = uuid.UUID(bytes=resp.callUnaryResponse.callId)
+
+                if call_id in self.pending_calls and resp.callUnaryResponse.HasField("result"):
+                    self.pending_calls[call_id].set_result(resp.callUnaryResponse.result)
+                elif call_id in self.pending_calls and resp.callUnaryResponse.HasField("error"):
+                    remote_exc = RemoteException(str(resp.callUnaryResponse.error))
+                    self.pending_calls[call_id].set_exception(remote_exc)
+                else:
+                    logger.debug(f"received unexpected unary call")
+
+            elif resp.HasField("requestHandling"):
+                asyncio.create_task(self._handle_persistent_request(resp.requestHandling))
+                pass
+
+    async def _write_to_persistent_conn(self, writer: asyncio.StreamWriter):
+        while True:
+            msg = await self.pending_messages.get()
+            await write_pbmsg(writer, msg)
+
+    async def _handle_persistent_request(self, request):
+        assert request.proto in self.unary_handlers
```

Comment:
Together with @deniskamazur, we've agreed to change it to `logger.warning` (or similar)